### PR TITLE
Faults can modify the price of an item

### PR DESCRIPTION
**Pull request by RenechCDDA**
#### Summary
Features "Item faults can modify the value of an item"

#### Purpose of change
Well, I intended this to be used for electronic items but it turns out electronic damage isn't fault-based, but an item flag. Oops.
(Still, it was easy and it'll be valuable later, so why not?)

#### Describe the solution
Add a new fault field (double, default 1.0), modify the item's value when that fault is present. This means that faults modify the value by 1.0 (i.e. they have no effect) unless a value is specifically given to them.

When the game prices an item it goes over all the faults and applies the price mod for each, so an engine that is missing its drive belt *and* has a bad pump *and* has an expired filter *and* has an immobiliser has an ultimate price mod of (0.7\*0.7\*0.7\*0.25 =) ~8.57% of its normal value, basically only worth scrap minus the labor cost of taking it apart!

#### Describe alternatives you've considered
It would be nice if it scaled with the value of some faults (e.g. fouling buildup) but those are stored as item vars despite being associated with faults. This could (maybe?) be unified at some point, but out of scope for this little change.

#### Testing
image

#### Additional context
Price modifiers were chosen somewhat arbitrarily, but based on my layperson's mindset of "How problematic is this for me to resolve". An engine that doesn't start at all (50% price mod) or has some sort of security system (25% price mod) is basically unapproachable without mechanical knowledge, hence the penalties for those are more severe.

### CODE REVIEW

**Comment by TheShadowFerret** [comment]
How would merchants/you know items from the opposing party are faulty, other than the worn drive belt or immobilizer🤔

File: data/json/faults/faults_vehicles.json [hunk 1]
```diff
@@ -29,6 +32,7 @@
     "name": { "str": "Faulty diesel pump" },
     "description": "Required to pump and pressurize diesel from a vehicles tank.",
```
Comment (TheShadowFerret) [inline data/json/faults/faults_vehicles.json:33]:
```suggestion
    "description": "Required to pump and pressurize diesel from a vehicle's tank.",
```

File: data/json/faults/faults_vehicles.json [hunk 2]
```diff
@@ -53,6 +59,7 @@
     "name": { "str": "Faulty fuel pump" },
     "description": "Required to pump gasoline from a vehicles tank.",
```
Comment (TheShadowFerret) [inline data/json/faults/faults_vehicles.json:60]:
```suggestion
    "description": "Required to pump gasoline from a vehicle's tank.",
```

**Comment by RenechCDDA** [comment]
> How would merchants/you know items from the opposing party are faulty, other than the worn drive belt or immobilizer(which are perceptible without testing the device)?🤔
> 
> (Handwaving to prevent cheese is fine to me, for the record)

I presume like any competent merchant they would examine the goods and ask for a demonstration. If the engine doesn't start, well... heh, they're not going to offer you a very good price. (CBMs are more like a parts kit that's been unpacked, there's an element of wariness that some unknown part might be missing.)

There might be some faults which should normally be 'hidden', but given the player can identify faults on sight regardless of their technical knowledge, I think it is fine for mercantile NPCs to have this ability too. If someone wants to wrangle fault visibility in the future, they can be my guest.

**Pull request merged**